python/cuda_parallel/cuda/parallel/experimental/algorithms/reduce.py

```diff
@@ -85,7 +109,9 @@ def __init__(
         if error != enums.CUDA_SUCCESS:
             raise ValueError("Error building reduce")
 
-    def __call__(self, temp_storage, d_in, d_out, num_items: int, h_init: np.ndarray):
+    def __call__(
+        self, temp_storage, d_in, d_out, num_items: int, h_init: np.ndarray, stream=None
```

Comment:
I don't have strong opinion in this, just FYI: in some places in `cuda.core` we have the following signature
```python
def func(..., *, stream):
```
so that we enforce `stream` to be passed as a kw arg, but has no default value. The consideration back then was to ensure users to pass a stream explicitly.

Reply:
I think given the above discussion, this should be fine for now, but happy to make this change if you prefer.

Reply:
yeah don't worry about it for now